pkg/controller.v1/common/job.go

```diff
@@ -95,7 +95,11 @@ func (jc *JobController) ReconcileJobs(
 	}
 
 	oldStatus := jobStatus.DeepCopy()
-	if commonutil.IsSucceeded(jobStatus) || commonutil.IsFailed(jobStatus) {
+	jobSuspended, err := jc.Controller.JobSuspended(job)
+	if err != nil {
+		return err
+	}
+	if commonutil.IsSucceeded(jobStatus) || commonutil.IsFailed(jobStatus) || (jobSuspended != nil && *jobSuspended) {
```

Comment:
My concern here is `Suspend` is just a transition state, should we delete all the pods or just the active ones, leaving the completed pods(succeeded/failed) as they are.

Reply:
Yes, the current implementation is consistent with batch/job.